File: cpp_solutions/3068.cpp
class Solution {
public:
    long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {
        long long sum = 0;
        long long cnt = 0;
        long long sacrifice = INT_MAX;
        for (long long n : nums) {
            sum += max(n ^ k, n);
            cnt += (n ^ k) > n;
            sacrifice = min(sacrifice, abs(n - (n ^ k)));
        }
        return sum - (cnt % 2 ? sacrifice : 0);
    }
};

File: rust_solutions/3068.rs
impl Solution {
    pub fn maximum_value_sum(nums: Vec<i32>, k: i32, edges: Vec<Vec<i32>>) -> i64 {
        let mut sum: i64 = 0;
        let mut cnt: i32 = 0;
        let mut sacrifice: i64 = i64::MAX;

        for &num in &nums {
            let num = num as i64;
            let num_xor = num ^ k as i64;
            sum += num.max(num_xor);

            cnt += (num_xor > num) as i32;
            sacrifice = sacrifice.min((num - num_xor).abs());
        }
        if cnt % 2 == 1 {
            return sum - sacrifice;
        }
        return sum;
    }
}
